types/field_type.go

```diff
@@ -308,6 +309,15 @@ func MergeFieldType(a byte, b byte) byte {
 	return fieldTypeMergeRules[ia][ib]
 }
 
+// mergeTypeFlag merges two MySQL type flag to a new one
+// currently only NotNullFlag is checked
+// todo more flag need to be checked, for example: UnsignedFlag
+func mergeTypeFlag(a, b uint) uint {
+	ret := a
```

Comment:
`return a & (b & mysql.NotNullFlag | ^mysql.NotNullFlag)` is ok ?

Reply:
Yes.